doc/design/eventdriven-callbacks.md

```diff
@@ -0,0 +1,171 @@
+## Name proposals
+
+ - ReactAl = React And Listen
+ - Reactor
+ - Eventler
+
+
+## Usage
+
+### General
+The usage should be similar to the _WaitSet_ with a key difference - it should 
+be **event driven** and not a mixture of event and state driven, depending on
+which event is attached, like in the _WaitSet_.
+
+It should have the following behavior:
+
+ - Whenever an event occurs the corresponding callback should be called **once**
+    immediately.
+ - If an event occurs multiple times before the callback was called, the callback
+    should be called **once**.
+ - If an event occurs while the callback is being executed the callback should be 
+    called again **once**.
```

Comment:
Nicely pointed out :sunflower:  However, we should define a name for this behaviour. I'm quite font of the name `CatchupPolicy`. What you are describing is probably "ASAP" and "N_TO_1" behaviour. As the `WaitSet` is the sibling of ReactAl, we could add such a description there, too.

Reply:
- ASAP describes when you react (as soon as possible) and not how many times.
- N_TO_1 is when you channel N notifications to one recipient which is the Reactor Pattern and we have two solutions for it, the WaitSet and the upcoming "ReactAl"
- `CatchupPolicy` meant in the timer, if you were activated 5 times and didn't call the callback once then call the callback 5 times and then only every time you were activated.

We are following here the typical Reactive Programming approach which means an event causes a callback (once). We  could think of a name for this kind of behavior but it would most likely not end up in some C++ code since this behavior is tightly coupled to the ReactAl and changing the behavior via template/ctor arg would make the ReactAl even more complex as it is right now.

But if you have a nice fitting name I would welcome it very much!

Reply:
Ok, I think we talked past each other a bit here. I didn't mean to introduce an enum here, rather having some policies like the DDS QoS one. Then we could formally write down which reactive pattern fulfill which policy to make it easier for the user to choose and to have a clear differentiation for use-cases. Still searching for good names, though :-/

Reply:
Added todo to the issue " Create use case based comparison table between WaitSet and callback proposal" where this will be handled when the proposal is implemented since then we exactly know how the ReactAl will behave.